modules/compiler/utils/source/replace_local_module_scope_variables_pass.cpp

```diff
@@ -236,12 +236,13 @@ PreservedAnalyses compiler::utils::ReplaceLocalModuleScopeVariablesPass::run(
     // then add our element type to the struct
     structElementTypes.push_back(memberType);
 
-    // update the offset, taking into account any array length
-    unsigned int totalSize = alignment;
-    while (memberType->isArrayTy()) {
-      totalSize *= memberType->getArrayNumElements();
-      memberType = memberType->getArrayElementType();
+    // update the offset based on the type's size
+    if (dl.getTypeAllocSize(memberType).isScalable()) {
+      // Not an assert because this can happen in user-supplied IR
+      llvm::report_fatal_error(
+          "Scalable types in local memory are not yet supported");
     }
+    unsigned int totalSize = dl.getTypeAllocSize(memberType).getFixedValue();
```

Comment:
I don't know if `dl.getTypeAllocSize()` caches its results so that for consecutive calls, it does not do the same work of going over struct members repeatedly. If it doesn't do that, then this looks good to me. If it does, it might be better to store the result in a local variable. Will tentatively approve and leave it to you to determine whether this is an issue to change; looks good to me otherwise.